src/test/java/bio/terra/service/snapshotbuilder/query/table/DomainOccurrenceTest.java

```diff
@@ -0,0 +1,61 @@
+package bio.terra.service.snapshotbuilder.query.table;
+
+import static bio.terra.service.snapshotbuilder.utils.CriteriaQueryBuilderTest.assertQueryEquals;
+
+import bio.terra.common.category.Unit;
+import bio.terra.model.SnapshotBuilderConcept;
+import bio.terra.model.SnapshotBuilderDomainOption;
+import bio.terra.service.snapshotbuilder.query.FieldVariable;
+import bio.terra.service.snapshotbuilder.query.SqlRenderContext;
+import bio.terra.service.snapshotbuilder.query.SqlRenderContextProvider;
+import bio.terra.service.snapshotbuilder.utils.constants.ConditionOccurrence;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+@Tag(Unit.TAG)
+class DomainOccurrenceTest {
+
+  @ParameterizedTest
+  @ArgumentsSource(SqlRenderContextProvider.class)
+  void testDomainOccurrence(SqlRenderContext context) {
+    SnapshotBuilderDomainOption domainOption = new SnapshotBuilderDomainOption();
+    domainOption.root(new SnapshotBuilderConcept()).tableName("table");
+
+    DomainOccurrence domainOccurrence = DomainOccurrence.forPrimary(domainOption);
+
+    assertQueryEquals("table AS t", domainOccurrence.renderSQL(context));
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(SqlRenderContextProvider.class)
+  void testLeftJoinOn(SqlRenderContext context) {
+    SnapshotBuilderDomainOption domainOption = new SnapshotBuilderDomainOption();
+    domainOption
+        .root(new SnapshotBuilderConcept())
+        .tableName(ConditionOccurrence.TABLE_NAME)
```

Comment:
These names don't have meaning outside this test so it can be changed to declare them locally inside this test class, or this test method.